src/Products/Editor/Controllers/EditorApiController.cs

```diff
@@ -255,34 +263,85 @@ public HttpResponseMessage SaveFile(EditDocumentRequest postedData)
                     }
 
-                    using (EditableDocument beforeEdit = editor.Edit(editOptions))
+                    if (saveOptions is WordProcessingSaveOptions)
                     {
-                        EditableDocument htmlContentDoc = EditableDocument.FromMarkup(htmlContent, null);
-                        dynamic saveOptions = GetSaveOptions(guid);
-                        saveOptions.Password = password;
+                        saveOptions.EnablePagination = true;
+                    }
 
+                    using (FileStream outputStream = File.Create(tempPath))
+                    {
+                        editor.Save(htmlContentDoc, outputStream, saveOptions);
+                    }
+                }
+
+                if (File.Exists(saveFilePath))
+                {
+                    File.Delete(saveFilePath);
+                }
+
+                File.Move(tempPath, saveFilePath);
+
+                LoadDocumentEntity loadDocumentEntity = LoadDocument(saveFilePath, password);
+
+                // return document description
+                return Request.CreateResponse(HttpStatusCode.OK, loadDocumentEntity);
+            }
+            catch (Exception ex)
+            {
+                // set exception message
+                return Request.CreateResponse(HttpStatusCode.InternalServerError, new Resources().GenerateException(ex, postedData.getPassword()));
+            }
+        }
+
+        /// <summary>
+        /// Load document description
+        /// </summary>
+        /// <param name="postedData">Post data</param>
+        /// <returns>Document info object</returns>
+        [HttpPost]
+        [Route("createFile")]
+        public HttpResponseMessage CreateFile(EditDocumentRequest postedData)
+        {
+            try
+            {
+                string htmlContent = postedData.getContent();
+                string guid = postedData.GetGuid();
+                string saveFilePath = Path.Combine(globalConfiguration.GetEditorConfiguration().GetFilesDirectory(), guid);
+                string tempFilename = Path.GetFileNameWithoutExtension(saveFilePath) + "_tmp";
+                string tempPath = Path.Combine(Path.GetDirectoryName(saveFilePath), tempFilename + Path.GetExtension(saveFilePath));
+
+                File.Create(saveFilePath).Dispose();
+
+                using (EditableDocument newFile = EditableDocument.FromMarkup(htmlContent, null))
+                {
+                    using (GroupDocs.Editor.Editor editor = new GroupDocs.Editor.Editor(saveFilePath))
+                    {
+                        dynamic saveOptions = this.GetSaveOptions(saveFilePath);
                         if (saveOptions is WordProcessingSaveOptions)
                         {
-                            saveOptions.EnablePagination = true;
+                            // TODO: leads to exception
+                            //saveOptions.EnablePagination = true;
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/groupdocs-editor/GroupDocs.Editor-for-.NET-MVC/pullRequest?prid=6003709)